src/sentry/db/models/manager/base_query_set.py

```diff
@@ -4,12 +4,42 @@
 from django.db import router
 from django.db.models import QuerySet
 
+from sentry.signals import post_update
+
 
 class BaseQuerySet(QuerySet, abc.ABC):
-    # XXX(dcramer): we prefer values_list, but we can't disable values as Django uses it
-    # internally
-    # def values(self, *args, **kwargs):
-    #     raise NotImplementedError('Use ``values_list`` instead [performance].')
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._send_post_update_signal = False
+
+    def enable_post_update_signal(self, enable: bool) -> "BaseQuerySet":
+        """
+        Enables sending a `post_update` signal after this queryset runs an update command. Note that this is less
+        efficient than just running the update. To get the list of group ids affected, we first run the query to
+        fetch the rows we want to update, then run the update using those ids.
+        """
+        qs = self.all()
+        qs._send_post_update_signal = enable
+        return qs
+
+    def _clone(self) -> "BaseQuerySet":
```

Comment:
This is a bit hacky, but if we don't override clone here then we'll lose track of the value of `_send_post_update_signal` between clones